File: Low Level Design (LLD)/Problems/ParkingLot/Java/model/ParkingSlot.java
package model;

public class ParkingSlot {
    private final int slotNumber;
    private final int floorNumber;
    private final VehicleType type;
    private boolean isOccupied;
    private Vehicle currentVehicle;

    public ParkingSlot(int slotNumber, int floorNumber, VehicleType type) {
        this.slotNumber = slotNumber;
        this.floorNumber = floorNumber;
        this.type = type;
        this.isOccupied = false;
    }

    public boolean canFitVehicle(Vehicle vehicle) {
        return !isOccupied && vehicle.getType() == type;
    }

    public void assignVehicle(Vehicle vehicle) {
        this.currentVehicle = vehicle;
        this.isOccupied = true;
    }

    public void removeVehicle(Vehicle vehicle) {
        this.currentVehicle = null;
        this.isOccupied = false;
    }

    public int getSlotNumber() {
        return slotNumber;
    }

    public int getFloorNumber() {
        return floorNumber;
    }

    public Vehicle getCurrentVehicle() {
        return currentVehicle;
    }
}

File: Low Level Design (LLD)/Problems/02. ParkingLot/Cpp/model/src/ParkingSlot.cpp
#include "../include/ParkingSlot.h"

ParkingSlot::ParkingSlot(int slotNumber, int floorNumber, VehicleType type) 
    : slotNumber(slotNumber), floorNumber(floorNumber), type(type),
    isOccupied(false), currentVehicle(nullptr) {}

bool ParkingSlot::canFitVehicle(Vehicle* vehicle) {
    return !isOccupied && vehicle->getType() == type;
}

void ParkingSlot::assignVehicle(Vehicle* vehicle) {
    currentVehicle = vehicle;
    isOccupied = true;
}

void ParkingSlot::removeVehicle(Vehicle* vehicle) {
    currentVehicle = nullptr;
    isOccupied = false;
}

int ParkingSlot::getSlotNumber() const {
    return slotNumber;
}

int ParkingSlot::getFloorNumber() const {
    return floorNumber;
}

Vehicle* ParkingSlot::getCurrentVehicle() const {
    return currentVehicle;
}